src/ashell/comms-uart.c

```diff
@@ -523,13 +521,25 @@ void acm()
     k_fifo_init(&data_queue);
     k_fifo_init(&avail_queue);
 
+    // Wait until the main loop is running before starting the javascript
+    if (k_sem_take(&mainloop_sem, K_FOREVER) == 0)
+    {
+        ashell_run_boot_cfg();
+    }
+    else {
+        ERR_PRINT("Main loop failed to start\n");
+    }
+
 #ifdef CONFIG_UART_LINE_CTRL
     uint32_t dtr = 0;
 
     while (1) {
         uart_line_ctrl_get(dev_upload, LINE_CTRL_DTR, &dtr);
         if (dtr)
             break;
+        // Sleep is needed to allow the javascript in boot.cfg to run
+        // while we wait for the connection.
+        k_sleep(2000);
```

Comment:
No.  This while loop is waiting for the WebUSB or ACM connection, hence the while(1).  The sleep is so that it doesn't eat all the cycles, preventing the javascript from running.